### PR TITLE
fix propertypane header

### PR DESCRIPTION
**Pull request by ram-primathon**
## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/ads/EditableText.tsx [hunk 1]
```diff
@@ -132,10 +132,7 @@ const TextContainer = styled.div<{
 
   &&& .${BlueprintClasses.EDITABLE_TEXT} {
     overflow: hidden;
-    height: ${(props) => props.theme.spaces[14] + 1}px;
-    padding: ${(props) => props.theme.spaces[4]}px
-      ${(props) => props.theme.spaces[5]}px;
-    width: calc(100% - 40px);
+    width: 100%;
```
Comment (hetunandu) [inline app/client/src/components/ads/EditableText.tsx:135]: Why did we remove this? Api pane styles are affected by this

File: app/client/src/components/ads/EditableText.tsx [hunk 2]
```diff
@@ -242,7 +238,7 @@ export const EditableText = (props: EditableTextProps) => {
       const error = errorMessage ? errorMessage : false;
       if (!error && finalVal !== "") {
         setLastValidValue(finalVal);
-        onTextChanged && onTextChanged(finalVal);
+        // onTextChanged && onTextChanged(finalVal);
```
Comment (hetunandu) [inline app/client/src/components/ads/EditableText.tsx:241]: Why is this not necessary anymore

File: app/client/src/pages/Editor/PropertyPaneTitle.tsx [hunk 3]
```diff
@@ -16,19 +14,21 @@ import { BindingText } from "pages/Editor/APIEditor/Form";
...
+import styled, { theme } from "constants/DefaultTheme";
 import { ControlIcons } from "icons/ControlIcons";
 import { FormIcons } from "icons/FormIcons";
 import { copyWidget, deleteSelectedWidget } from "actions/widgetActions";
+import { AnyStyledComponent } from "styled-components";
 
 const CopyIcon = ControlIcons.COPY_CONTROL;
 const DeleteIcon = FormIcons.DELETE_ICON;
-const Wrapper = styled.div`
+const Wrapper = styled.div<{ isPropertyTitle: boolean }>`
```
Comment (hetunandu) [inline app/client/src/pages/Editor/PropertyPaneTitle.tsx:25]: rename to `isPanelTitle`

File: app/client/src/pages/Editor/PropertyPaneTitle.tsx [hunk 4]
```diff
@@ -107,50 +144,70 @@ const PropertyPaneTitle = memo((props: PropertyPaneTitleProps) => {
...
+        >
+          <DeleteIcon
+            className="t--delete-widget"
+            width={16}
+            height={16}
+            color={theme.colors.paneSectionLabel}
+            onClick={handleDelete}
+          />
+        </Tooltip>
+      )}
```
Comment (hetunandu) [inline app/client/src/pages/Editor/PropertyPaneTitle.tsx:209]: Not the correct way to handle this. You need to make this composable from outside

File: app/client/src/pages/Editor/PropertyPaneTitle.tsx [hunk 5]
```diff
@@ -107,50 +144,70 @@ const PropertyPaneTitle = memo((props: PropertyPaneTitleProps) => {
...
-          isEditingDefault={isNew}
-          onBlur={updateTitle}
-          hideEditIcon
-          className="t--propery-page-title"
-          savingState={updating ? SavingState.STARTED : SavingState.NOT_STARTED}
-          fill
-          underline
-        />
-        {updating && <Spinner size={16} />}
+  return props.widgetId || props.updatePropertyTitle ? (
```
Comment (hetunandu) [inline app/client/src/pages/Editor/PropertyPaneTitle.tsx:147]: `updatePropertyTitle` is a wrong way to handle this

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/components/ads/EditableText.tsx (hunk 2 above)
Comment (ram-primathon) [reply app/client/src/components/ads/EditableText.tsx:241]: this was responsible to start loader on enter of every character

File: app/client/src/pages/Editor/PropertyPaneTitle.tsx (hunk 3 above)
Comment (ram-primathon) [reply app/client/src/pages/Editor/PropertyPaneTitle.tsx:25]: done

File: app/client/src/components/ads/EditableText.tsx (hunk 2 above)
Comment (hetunandu) [reply app/client/src/components/ads/EditableText.tsx:241]: Revert this change. I have already fixed that on the main branch

File: app/client/src/pages/Editor/PropertyPaneTitle.tsx (hunk 4 above)
Comment (ram-primathon) [reply app/client/src/pages/Editor/PropertyPaneTitle.tsx:209]: added 4 different propps for each

Comment (hetunandu) [reply app/client/src/pages/Editor/PropertyPaneTitle.tsx:209]: You need to make it composable. What this means it that you can add children from outside if you need more icons. 
```
<PropertyPaneTitle
   ....
   actions={(
     <DeleteIcon />
     <CopyIcon/>
   )}
/>
```

File: app/client/src/pages/Editor/PropertyPaneTitle.tsx (hunk 5 above)
Comment (ram-primathon) [reply app/client/src/pages/Editor/PropertyPaneTitle.tsx:147]: added a props 

isPanelTitle

File: app/client/src/components/ads/EditableText.tsx (hunk 1 above)
Comment (ram-primathon) [reply app/client/src/components/ads/EditableText.tsx]: revert back

File: app/client/src/components/ads/EditableText.tsx (hunk 2 above)
Comment (ram-primathon) [reply app/client/src/components/ads/EditableText.tsx]: done

File: app/client/src/pages/Editor/PropertyPaneTitle.tsx [hunk 6]
```diff
@@ -37,33 +38,68 @@ const Wrapper = styled.div`
...
   title: string;
   widgetId?: string;
   widgetType?: WidgetType;
   onClose: () => void;
+  updatePropertyTitle?: (title: string) => void;
+  onBackClick?: () => void;
+  hideCopyIcon?: boolean;
+  hideDeleteIcon?: boolean;
+  hideHelpIcon?: boolean;
+  hideCloseIcon?: boolean;
```
Comment (hetunandu) [inline app/client/src/pages/Editor/PropertyPaneTitle.tsx:101]: Make this composable

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/pages/Editor/PropertyPaneTitle.tsx (hunk 4 above)
Comment (ram-primathon) [reply app/client/src/pages/Editor/PropertyPaneTitle.tsx]: How are we suppose to bind click method with it?

**Review by hetunandu — APPROVED** [review]

**Pull request merged**